pkg/pg/collectors.go

```diff
@@ -366,3 +327,426 @@ func WaitEvents(pgPool *pgxpool.Pool) func(ctx context.Context, state *agent.Met
+			{Current: &tempFiles, Previous: &state.Cache.PGDatabase.TempFiles, Metric: metrics.PGTempFiles},
+			{Current: &tempBytes, Previous: &state.Cache.PGDatabase.TempBytes, Metric: metrics.PGTempBytes},
+			{Current: &deadlocks, Previous: &state.Cache.PGDatabase.Deadlocks, Metric: metrics.PGDeadlocks},
+		}
+
+		mappingsFloat := []MetricMapping[float64]{
+			{Current: &idleInTransactionTime, Previous: &state.Cache.PGDatabase.IdleInTransactionTime, Metric: metrics.PGIdleInTransactionTime},
+		}
+
+		if err := EmitDeltaMetricsMap(state, mappingsInt); err != nil {
+			return err
+		}
+		if err := EmitDeltaMetricsMap(state, mappingsFloat); err != nil {
+			return err
+		}
+
+		// Update cache for next iteration
+		state.Cache.PGDatabase = agent.PGDatabase{
+			BlksRead:              blksRead,
+			BlksHit:               blksHit,
+			TuplesReturned:        tupReturned,
+			TuplesFetched:         tupFetched,
+			TuplesInserted:        tupInserted,
+			TuplesUpdated:         tupUpdated,
+			TuplesDeleted:         tupDeleted,
+			TempFiles:             tempFiles,
+			TempBytes:             tempBytes,
+			Deadlocks:             deadlocks,
+			IdleInTransactionTime: idleInTransactionTime,
+			Timestamp:             time.Now(),
+		}
+
+		return nil
+	}
+}
+
+const PGStatUserTablesQuery = `
+/*dbtune*/
+SELECT JSON_OBJECT_AGG(
+	CONCAT(relname, '_', relid),
+	JSON_BUILD_OBJECT(
+        'last_autovacuum',last_autovacuum,
+        'last_autoanalyze',last_autoanalyze,
+        'autovacuum_count',autovacuum_count,
+        'autoanalyze_count',autoanalyze_count,
+        'n_live_tup',n_live_tup,
+        'n_dead_tup',n_dead_tup,
+        'n_mod_since_analyze',n_mod_since_analyze,
+        'n_ins_since_vacuum',n_ins_since_vacuum,
+        'seq_scan',seq_scan,
+        'seq_tup_read',seq_tup_read,
+        'idx_scan',idx_scan,
+        'idx_tup_fetch',idx_tup_fetch
+    )
+)
+as stats
+FROM pg_stat_user_tables
+as f
+`
+
+// PGStatUserTables collects statistics from pg_stat_user_tables, computes deltas for various metrics per table,
+// and emits them as metrics for monitoring purposes. It tracks metrics such as autovacuum counts, live/dead tuples,
+// and sequential/index scans, providing insights into table activity and performance over time.
+func PGStatUserTables(pgPool *pgxpool.Pool) func(ctx context.Context, state *agent.MetricsState) error {
+	return func(ctx context.Context, state *agent.MetricsState) error {
+		var jsonResult string
+		err := pgPool.QueryRow(ctx, PGStatUserTablesQuery).Scan(&jsonResult)
+		if err != nil {
+			if err.Error() == "can't scan into dest[0]: cannot scan NULL into *string" {
+				return errors.New("pg_stat_user_tables returned no data, likely this means there are no user tables in the database")
+			}
+			return err
+		}
+
+		var tableStats map[string]utils.PGUserTables
+		err = json.Unmarshal([]byte(jsonResult), &tableStats)
+		if err != nil {
+			return err
+		}
```

Comment:
@mbocevski not for now but this is also likely something we want to be able to update from the frontend, i.e. the user can select which tables we are allowed to query.